app/modules/api/controllers.py

```diff
@@ -92,4 +92,47 @@ def page_api_bans():
 		else:
 			return jsonify({"error": "No Ckey Provided!"})
 	except Exception as E:
-		return jsonify({"error": "Unknown!"})
\ No newline at end of file
+		return jsonify({"error": "Unknown!"})
+
+@bp_api.route("/cross_cargo")
+def page_api_cross_cargo():
+	try:
+		d = {}
+
+		for server in cfg.SERVERS:
+			if server["open"]:
+				try:
+					d[server["id"]] = util.fetch_server_icn(server["id"])
+				except Exception as E:
+					d[server["id"]] = {"error": str(E)}
+
+		return jsonify(d)
+
+	except Exception as E:
+		return jsonify({"error": str(E)})
+
+@bp_api.route("/cross_cargo/purchase")
+def page_api_cross_cargo_purchase():
+	comms_key = request.args.get('comms_key', type=str, default="")
+	if comms_key == cfg.PRIVATE["comms_key"]:
```

Comment:
I'm concerned about someone trying to bruteforce the comms key. Is there a way to ratelimit this function?

Reply:
Flask-Limiter package, but really this is unneeded